main.py

```diff
@@ -272,12 +270,22 @@ async def get_prefix(self, message: discord.Message) -> Union[List[str], str]:
                 "SELECT prefix FROM internal_prefix WHERE snowflake_id=$1",
                 snowflake_id,
             )
-            prefix = "uwu " if data is None else data["prefix"]
+            prefix = self._default_prefix if data is None else data["prefix"]
             self.existing_prefix[snowflake_id] = prefix
 
+        return prefix
+
+    async def get_prefix(self, message: discord.Message) -> Union[List[str], str]:
+        """
+        Escape special characters in prefix, then compile it as regular expression using case insensivity flag (yes, i
+        know i could compile them in cache but has anyone asked?). Try matching the beginning of message content using
+        regex. If match found, return match group 0 which will be just the prefix itself. Otherwise return empty list.
+        """
+        prefix = await self._prefix_for_message(message)
+
         if match := re.match(re.escape(prefix), message.content, flags=re.I):
             return match[0]
-        return prefix
+        return []
```

Comment:
why do you want it to return `[]` so much

Reply:
because get_prefix is supposed to test message for prefix. in this case nothing matched

Reply:
it really doesn't, the point of get_prefix, is to get a prefix for the bot, for the message is an additional feature, it really shouldn't come down to, oh I can't get a prefix from this message, return nothing

Reply:
and no this isnt logic error, you're just adding your own definition on what get_prefix is suppose to do, which I disagree on.